src/textual/widgets/_loading_indicator.py

```diff
@@ -0,0 +1,63 @@
+from __future__ import annotations
+
+from time import time
+
+from rich.console import RenderableType
+from rich.style import Style
+from rich.text import Text
+
+from ..color import Gradient
+from ..widget import Widget
+
+
+class LoadingIndicator(Widget):
+    """Display an animated loading indicator."""
+
+    COMPONENT_CLASSES = {"loading-indicator--dot"}
+
+    DEFAULT_CSS = """
+    LoadingIndicator {
+        width: 100%;
+        height: 100%;
+        content-align: center middle;
+    }
+
+    LoadingIndicator > .loading-indicator--dot {
+        color: $accent;
+    }
+
+    """
+
+    def on_mount(self) -> None:
+        self.start_time = time()
+        self.auto_refresh = 1 / 16
+
+    def render(self) -> RenderableType:
+        elapsed = time() - self.start_time
+        speed = 0.8
+        dot = "\u25CF"
```

Comment:
I think I can guarantee that folk will start asking how they change the character. I think ideally it should be possible for the dev to set it when constructing the widget. If not that it should be possible for someone to modify it in a subclass with minimal effort.

Reply:
I'm going to leave it for the time being. The problem with offering everything as an option is that you accumulate a lot of arguments, and you have to handle the edge cases. I'll leave in until we know for certain the demand is there.